### PR TITLE
Cleanup onCreateNode in the gatsby-node file

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
## What kind of change does this PR introduce?
After looking at the [`onCreateNode` function in the `gatsby-node` file](https://github.com/kentcdodds/kentcdodds.com/blob/5fcb4b2ad9e9cc55fda049faa08ef2be913c6afd/gatsby-node.js#L288-L443) of @kentcdodds' [personal website](https://github.com/kentcdodds/kentcdodds.com), I thought it would be a good idea to refactor our homepages `gatsby-node` file and put all mutations and default-values inside of this file's `onCreateNode` function.
This looks to me like a good idea so the components that are using those fields don't have to do it themselves and we don't have to repeat those steps in all the components.

## What is the current behavior?
Every component has to foresee their own default values or mutations to fields from within the corresponding Markdown file.

## What is the new behavior?
The `onCreateNode` function in the `gatsby-node` file takes care of foreseeing all default values and mutations so the components just have to query them and show them.

## What steps did you take to test this?
Go through all pages (blogposts, docs, jobs & legal pages) to see if everything is still working as it should be.

## Checklist
- [ ] Documentation N/A
- [x] Testing
- [x] Ready to be merged
- [ ] Added myself to contributors table N/A

### CODE REVIEW

**Comment by SaraVieira** [comment]
Hey!

Sorry can you fix the conflicts? :(

**Comment by MichaelDeBoey** [comment]
@SaraVieira I've fixed the merge conflicts 🙂

**Comment by MichaelDeBoey** [comment]
@SaraVieira I've fixed the merge conflicts 🙂

File: packages/homepage/src/templates/docs.js [hunk 1]
```diff
@@ -211,17 +211,20 @@ const Description = styled.p`
 // eslint-disable-next-line
 export default class Docs extends React.Component {
```
Comment (SaraVieira) [inline packages/homepage/src/templates/docs.js]: Can you make this a functional component? Maybe after this one is merged

File: packages/homepage/src/templates/docs.js [hunk 2]
```diff
@@ -263,33 +264,32 @@ export default class Docs extends React.Component {
 
 export const pageQuery = graphql`
   query Docs($slug: String!) {
-    allMarkdownRemark(
-      filter: { fields: { slug: { regex: "/docs/" } } }
-      sort: { fields: [fileAbsolutePath], order: ASC }
+    allDocs: allMarkdownRemark(
+      filter: { fileAbsolutePath: { regex: "/docs/" } }
+      sort: { fields: [fileAbsolutePath], order: [ASC] }
```
Comment (SaraVieira) [inline packages/homepage/src/templates/docs.js]: What is the difference between `[ASC]` and `ASC`?

**Comment by SaraVieira** [comment]
All looks good! I am gonna merge and if you would like it would be nice to move all the `css` to the elements file

Thank you!

File: packages/homepage/src/templates/docs.js (hunk 2 above)
Comment (MichaelDeBoey) [reply packages/homepage/src/templates/docs.js]: No real difference in this case, just a bit cleaner to specify it for each sorted field I think 🙂

File: packages/homepage/src/templates/docs.js (hunk 1 above)
Comment (MichaelDeBoey) [reply packages/homepage/src/templates/docs.js]: Will do in a follow-up PR 🙂

**Comment by MichaelDeBoey** [comment]
@SaraVieira Will do that in a follow-up PR 🙂

**Pull request merged**